Signal/src/ViewControllers/Utils/MessageRecipientStatusUtils.swift

```diff
@@ -13,13 +13,30 @@ import Foundation
     case failed
 }
 
+// Our per-recipient status messages are "biased towards success"
+// and reflect the most successful known state for that recipient.
+//
+// Our per-message status messages are "biased towards failure"
+// and reflect the least successful known state for that message.
+//
+// Why?
+//
+// When showing the per-recipient status, we want to show the message
+// as "read" even if delivery failed to another recipient of the same
+// message.
+//
+// When showing the per-message status, we want to show the message
+// as "failed" if delivery failed to any recipient, even if another 
+// receipient has read the message.
 class MessageRecipientStatusUtils: NSObject {
     // MARK: Initializers
 
     @available(*, unavailable, message:"do not instantiate this class.")
     private override init() {
     }
 
+    // This method is per-recipient and "biased towards success".
+    // See comments above.
     public class func recipientStatus(outgoingMessage: TSOutgoingMessage,
```

Comment:
tldr; move the comments about bias to `recipientStatus(outgoingMessage:)` and remove the other comments.

NBD: As I understand this concept of "bias", we're talking about how to aggregate the status of the recipient(s) into a single status for the overall message.

When there is only a single recipient, there is no aggregation. It's simply the status of the single recipient. When there are multiple recipients we have some more complicated aggregation logic.  e.g. the message bubble footer text "Delivered" appears as soon as at least one person has been delivered the message.

As such, I found the comment introducing the concept of "bias" in single recipient status method confusing rather than helpful. In the group context, where we *have* aggregation logic, it makes sense to spell it out a bit more.

Reply:
That's not quite right.  I've elaborated the comment to be more clear why I speak of bias.

Reply:
Ah, so we also need the bias in order to do the right (usually) thing for per-recipient details on legacy messages.

Reply:
I guess it's subjective, but in my opinion we're applying bias in the per-message case too.